// ng2-semantic-ui/components/message/message.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var Message = (function () {
    function Message() {
        this.dismissible = true;
        this.onDismiss = new core_1.EventEmitter(false);
        this.dismissed = false;
        this.classes = "";
    }
    Message.prototype.dismiss = function () {
        this.dismissed = true;
        this.onDismiss.emit(this);
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Boolean)
    ], Message.prototype, "dismissible", void 0);
    __decorate([
        core_1.Output(), 
        __metadata('design:type', core_1.EventEmitter)
    ], Message.prototype, "onDismiss", void 0);
    __decorate([
        core_1.Input("class"), 
        __metadata('design:type', String)
    ], Message.prototype, "classes", void 0);
    Message = __decorate([
        core_1.Component({
            selector: 'sui-message',
            directives: [],
            template: "\n<div class=\"ui message {{ classes }}\" *ngIf=\"!dismissed\">\n    <i class=\"close icon\" *ngIf=\"dismissible\" (click)=\"dismiss()\"></i>\n    <ng-content></ng-content>\n</div>\n"
        }), 
        __metadata('design:paramtypes', [])
    ], Message);
    return Message;
}());
exports.Message = Message;
